_inc/connect-button.js

```diff
@@ -125,12 +125,21 @@ jQuery( document ).ready( function ( $ ) {
 		},
 		receiveData: function ( event ) {
 			if (
-				event.origin === jpConnect.jetpackApiDomain &&
-				event.source === jetpackConnectIframe.get( 0 ).contentWindow &&
-				event.data === 'close'
+				event.origin !== jpConnect.jetpackApiDomain ||
+				event.source !== jetpackConnectIframe.get( 0 ).contentWindow
 			) {
-				window.removeEventListener( 'message', this.receiveData );
-				jetpackConnectButton.handleAuthorizationComplete();
+				return;
+			}
+
+			switch ( event.data ) {
+				case 'close':
+					window.removeEventListener( 'message', this.receiveData );
```

Comment:
Maybe it would make sense to remove the event listener in both cases.

Reply:
Good catch.
However, in case of `wpcom_cookie` the whole page gets redirected, so I don't think we need this.

The `close` case needs this because the page will not necessarily reload.